docs/md-layout.js

```diff
@@ -0,0 +1,23 @@
+const GrayMatter = require('gray-matter');
+
+// makes mdx in next.js suck less by injecting necessary exports so that
+// the docs are still readable on github
+// (Shamelessly stolen from React Query docs)
+// (Which was shamelessly stolen from Expo.io docs)
+// @see https://github.com/tannerlinsley/react-query/blob/16b7d290c70639b627d9ada32951d211eac3adc3/docs/src/lib/docs/md-loader.js
+// @see https://github.com/expo/expo/blob/303cb7b689603223401c091c6a2e1e01f182d355/docs/common/md-loader.js
+
+module.exports = function addLayoutToMdx(source) {
+  const callback = this.async();
+
+  const { content, data } = GrayMatter(source);
```

Comment:
I suppose that this data comes from `YAML front matter` or different similar format. Is the metadata automatically injected into .mdx files generatred by api extractor, or for some articles we would need to edit it manually?

Reply:
The flow is like this now:
1. API extractor generates .json file
2. Our script generates .md files (not .mdx)
3. Our script also injects the front matter to the .md files